dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/ContentletTransformer.java

```diff
@@ -278,9 +279,19 @@ private static Object getObjectValue(Map<String, Object> originalMap, Field fiel
             value = originalMap.get(field.getFieldContentlet());
         }
 
-        //KeyValue objects must be returned as Maps
-        if(LegacyFieldTypes.KEY_VALUE.legacyValue().equals(field.getFieldType()) && value instanceof String ){
-            value = com.dotmarketing.portlets.structure.model.KeyValueFieldUtil.JSONValueToHashMap((String)value);
+        //KeyValue Fields must be returned as Maps
+        if (LegacyFieldTypes.KEY_VALUE.legacyValue().equals(field.getFieldType())
+                && value instanceof String) {
+            final String asString = value.toString();
+            try {
+                value = com.dotmarketing.portlets.structure.model.KeyValueFieldUtil
+                        .JSONValueToHashMap(asString);
+            } catch (Throwable e) {
+                Logger.warn(ContentletTransformer.class, () -> String
+                        .format("Failed to convert keyValue field `%s` to an actual json. An Empty will be returned instead. ",
+                                field.getVelocityVarName()));
+                value = new HashMap<>();
```

Comment:
if this value won't modified, use Collections.emptyMap()

Reply:
you got it. Done